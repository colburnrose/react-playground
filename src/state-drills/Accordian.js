import React from "react";

class Accordian extends React.Component {
  static defaultProps = {
    sections: [],
  };

  state = {
    currentIndex: 0,
  };

  handleButtonClick = (index) => {
    this.setState({ currentIndex: index });
  };

  renderSectionItems(currentIndex) {
    return this.props.sections.map((section, index) => (
      <li key={index}>
        <button type="button" onClick={() => this.handleButtonClick(index)}>
          {section.title}
        </button>
        {currentIndex === index && <p>{section.content}</p>}
      </li>
    ));
  }

  render() {
    return (
      <div>
        <ul>{this.renderSectionItems(this.state.currentIndex)}</ul>
      </div>
    );
  }
}

export default Accordian;
