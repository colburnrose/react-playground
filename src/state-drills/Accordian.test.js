import React from "react";
import ReactDOM from "react-dom";
import Accordian from "./Accordian";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`Accordians Component`, () => {
  const sections = [
    {
      title: "Section 1",
      content: "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
    },
    {
      title: "Section 2",
      content:
        "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!",
    },
    {
      title: "Section 3",
      content:
        "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?",
    },
  ];

  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders empty given no sections", () => {
    const wrapper = shallow(<Accordian />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders empty given no sections", () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
