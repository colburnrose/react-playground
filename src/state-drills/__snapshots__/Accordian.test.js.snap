// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordians Component renders empty given no sections 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordians Component renders empty given no sections 2`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
